src/messaging/CredentialApiMessageTypes.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { Attestation, Claim, Credential, CType, Quote } from '@kiltprotocol/core'
+import { DataUtils, SDKErrors } from '@kiltprotocol/utils'
+import * as Did from '@kiltprotocol/did'
+import { isHex } from '@polkadot/util'
+
+import {verifyMessageEnvelope} from './MessageEnvelope'
+import type {
+  IMessage,
+  MessageBody,
+} from '../types'
+
+/**
+     * Checks if the message body is well-formed.
+     *
+     * @param body The message body.
+     */
+export function verifyMessageBody(body: MessageBody): void {
```

Comment:
this could be a type guard asserting that we are dealing with a credential api message type

Reply:
You mean the name of the function should be different?

Reply:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards